vms/avm/network/network.go

```diff
@@ -116,71 +204,59 @@ func (n *network) AppGossip(ctx context.Context, nodeID ids.NodeID, msgBytes []b
 		return nil
 	}
 
-	if err := n.issueTx(tx); err == nil {
+	err = n.mempool.Add(tx)
+	if err == nil {
 		txID := tx.ID()
+		n.txPushGossiper.Add(tx)
+		if err := n.txPushGossiper.Gossip(ctx); err != nil {
+			n.ctx.Log.Error("failed to gossip tx",
+				zap.Stringer("txID", tx.ID()),
+				zap.Error(err),
+			)
+		}
 		n.gossipTxMessage(ctx, txID, msgBytes)
 	}
 	return nil
 }
 
-func (n *network) IssueTx(ctx context.Context, tx *txs.Tx) error {
-	if err := n.issueTx(tx); err != nil {
+// IssueTx attempts to add a tx to the mempool, after verifying it. If the tx is
+// added to the mempool, it will attempt to push gossip the tx to random peers
+// in the network using both the legacy and p2p SDK.
```

Comment:
Would it be better to just key off of support based on the handshake?

Reply:
I think we already get `version` in the `Connected` message, so it may not be that bad but could be wrong.

Reply:
We use AppGossip for this, so we don't have the ability to select who exactly we are sending the messages to for push gossip. We could filter the pull gossip to only send it to newer nodes if we cared to optimize this during the time between this release and the network upgrade.